vxsandbox/tests/test_rlimiter.py

```diff
@@ -1,20 +1,90 @@
 """Tests for vxsandbox.rlimiter."""
 
+import resource
+
 from vumi.tests.helpers import VumiTestCase
 
-from vxsandbox import rlimiter
 from vxsandbox.rlimiter import SandboxRlimiter
 
 
 class TestSandboxRlimiter(VumiTestCase):
-    def test_script_name_dot_py(self):
-        self.patch(rlimiter, '__file__', 'foo.py')
-        self.assertEqual(SandboxRlimiter.script_name(), 'foo.py')
+    """
+    The actual spawning of the process is tested elsewhere.
+    """
+
+    def test_build_script(self):
+        """
+        The script contains a bunch of `ulimit` commands and an `exec`.
+        """
+        rlimiter = SandboxRlimiter({
+            resource.RLIMIT_CPU: (40, 60),
+            resource.RLIMIT_NOFILE: (15, 15),
+        }, [])
+        self.assertEqual(rlimiter.build_script(), "\n".join([
+            "#!/bin/bash",
+            "",
+            "# Set resource limits.",
+            "ulimit -St 40",
+            "ulimit -Ht 60",
+            "ulimit -Sn 15",
+            "ulimit -Hn 15",
+            "",
+            'exec "$@"',
+        ]))
+
+    def test_build_script_negative_rlimit(self):
+        """
+        On some systems, -1 means "no limit".
+        """
+        cpu_hard = resource.getrlimit(resource.RLIMIT_CPU)[1]
+        nofile_soft = resource.getrlimit(resource.RLIMIT_NOFILE)[0]
+        rlimiter = SandboxRlimiter({
+            resource.RLIMIT_CPU: (40, -1),
+            resource.RLIMIT_NOFILE: (-1, 15),
+        }, [])
+        self.assertEqual(rlimiter.build_script(), "\n".join([
+            "#!/bin/bash",
+            "",
+            "# Set resource limits.",
+            "ulimit -St 40",
+            "ulimit -Ht %s" % (cpu_hard,),
+            "ulimit -Sn %s" % (nofile_soft,),
+            "ulimit -Hn 15",
+            "",
+            'exec "$@"',
+        ]))
 
-    def test_script_name_dot_pyc(self):
-        self.patch(rlimiter, '__file__', 'foo.pyc')
-        self.assertEqual(SandboxRlimiter.script_name(), 'foo.py')
+    def test_build_script_no_RLIMIT_STACK(self):
```

Comment:
Do we want a separate test for MEMLOCK skipping, or should we declare this a general test for the skipping functionality?